nncf/common/graph/graph_matching.py

```diff
@@ -0,0 +1,236 @@
+"""
+ Copyright (c) 2019 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from itertools import chain, combinations
+from typing import Callable, List
+from typing import Dict
+
+import numpy as np
+import networkx as nx
+
+
+def powerset(iterable, min_r=1, max_r=None):
+    if not isinstance(iterable, list):
+        s = list(iterable)
+    else:
+        s = iterable
+    if max_r is None:
+        max_r = len(s)
+    return chain.from_iterable(combinations(s, r) for r in range(min_r, max_r + 1))
+
+
+class Expression:
```

Comment:
Please add docstring for classes, methods here and below.

Reply:
Done